app/services/weather_service.rb

```diff
@@ -13,6 +13,48 @@ def self.format_munchie(parse_m)
         }
   end
 
+  def self.get_destination_weather(lat, lon, time)
+    
+    response = conn.get("forecast.json?key=#{Rails.application.credentials.weather_api[:key]}&q=#{lat}, #{lon}&days=5")
+    first_parse = JSON.parse(response.body, symbolize_names: true)
+    parse = first_parse[:forecast][:forecastday]
+    weather_at_eta = eta_parse(parse, time)
+    # require 'pry';binding.pry
+  end
+  
+  def self.eta_parse(parse, time)
+    t = Time.now + time[:time]
+    rounded = Time.at((t.to_i / 3600.0).round * 3600)
+    date_time = rounded.strftime("%Y-%m-%d %H:%M")
+    date = date_time.split(' ').first
+    weather_at_eta = []
+    parse.each do |day|
+      if day[:date] == date
+        day[:hour].each do |hour|
+          if hour[:time] == date_time
```

Comment:
This doesn't quite work, but I'm on a new branch and gonna merge it then. This is almost there though. Have a little more stuff implemented to handle more edge cases.